File: UVa/UVa377.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);
  int n {};
  string line {};
  getline( cin, line );
  n = stoi( line );
  unordered_map<char, char> c24{ {'V', '0'}, {'U', '1'}, {'C', '2'}, {'D', 3} };
  cout << "COWCULATIONS OUTPUT\n";
  while( n -- ){
    long long A {}, B {};
    getline( cin, line );
    string::size_type sz = 0;
    for( auto& c : line ) c = c24[c];
    string t( line.rbegin(), line.rend() );
    A = stoll( t, &sz, 4 );
    getline( cin, line );
    for( auto& c : line ) c = c24[c];
    string t2( line.rbegin(), line.rend() );
    B = stoll( t2, &sz, 4 );
    for( int i = 0; i < 3; ++ i ){
      getline( cin, line );
      if( line == "A" ) B = A + B;
      else if( line == "L" ) B /= 4;
      else if( line == "R" ) B *= 4;
    }
    getline( cin, line );
    for( auto& c : line ) c = c24[c];
    string t3( line.rbegin(), line.rend() );
    long long C = stoll( t3, &sz, 4 );
    if( B == C ) cout << "YES\n";
    else cout << "NO\n";
  }
  cout << "END OF OUTPUT\n";
  return 0;
}

File: UVa/UVa377.py
import sys


def F(c):
    if c == 'V':
        return '0'
    elif c == 'U':
        return '1'
    elif c == 'C':
        return '2'
    elif c == 'D':
        return '3'

n = int(sys.stdin.readline())
print('COWCULATIONS OUTPUT')
while n:
    A, B = sys.stdin.readline(), sys.stdin.readline()
    ops = [0] * 3
    for i in range(3):
        ops[i] = sys.stdin.readline().strip()
    C = sys.stdin.readline()
    A = list(A)
    B = list(B)
    C = list(C)
    A = [F(a) for a in A]
    A = [a for a in A if a]
    A = ''.join(A)
    B = [F(b) for b in B]
    B = [b for b in B if b]
    B = ''.join(B)
    C = [F(c) for c in C]
    C = [c for c in C if c]
    C = ''.join(C)
    a = int(A, 4)
    b = int(B, 4)
    for op in ops:
        if op == 'A':
            b = a + b
        elif op == 'L':
            b = b * 4
        elif op == 'R':
            b = b // 4
    c = int(C, 4)
    if b == c:
        print('YES')
    else:
        print('NO')
    n -= 1
print('END OF OUTPUT')
